packages/axelar-wasm-std/src/msg_id/base_58_solana_event_index.rs

```diff
@@ -0,0 +1,362 @@
+        bytes
+    }
+
+    fn random_tx_digest() -> String {
+        bs58::encode(random_bytes()).into_string()
+    }
+
+    fn random_event_index() -> u32 {
+        rand::random()
+    }
+
+    #[test]
+    fn should_parse_msg_id() {
+        let res = Base58SolanaTxDigestAndEventIndex::from_str(
+            "4hHzKKdpXH2QMB5Jm11YR48cLqUJb9Cwq2YL3tveVTPeFkZaLP8cdcH5UphVPJ7kYwCUCRLnywd3xkUhb4ZYWtf5-0",
+        );
+        res.unwrap();
+
+        for _ in 0..1000 {
+            let tx_digest = random_tx_digest();
+            let event_index = random_event_index();
+            let msg_id = format!("{}-{}", tx_digest, event_index);
+
+            let res = Base58SolanaTxDigestAndEventIndex::from_str(&msg_id);
+            let parsed = res.unwrap();
+            assert_eq!(parsed.event_index, event_index);
+            assert_eq!(parsed.tx_digest_as_base58(), tx_digest.try_into().unwrap());
+            assert_eq!(parsed.to_string(), msg_id);
+        }
+    }
+
+    #[test]
+    fn should_not_parse_msg_id_with_wrong_length_base58_tx_digest() {
+        let tx_digest = random_tx_digest();
+        let event_index = random_event_index();
+
+        // too long
+        let msg_id = format!("{}{}-{}", tx_digest, tx_digest, event_index);
+        let res = Base58SolanaTxDigestAndEventIndex::from_str(&msg_id);
+        assert!(res.is_err());
+
+        // too short
+        let msg_id = format!("{}-{}", &tx_digest[0..30], event_index);
+        let res = Base58SolanaTxDigestAndEventIndex::from_str(&msg_id);
+        assert!(res.is_err());
+    }
+
+    #[test]
+    fn leading_ones_should_not_be_ignored() {
+        let tx_digest = random_tx_digest();
+        let event_index = random_event_index();
+
+        let res =
+            Base58SolanaTxDigestAndEventIndex::from_str(&format!("1{}-{}", tx_digest, event_index));
+        assert!(res.is_err());
+
+        let res = Base58SolanaTxDigestAndEventIndex::from_str(&format!(
+            "11{}-{}",
+            tx_digest, event_index
+        ));
+        assert!(res.is_err());
+    }
+
+    #[test]
+    fn should_not_parse_msg_id_with_correct_length_base58_but_wrong_length_hex() {
+        // this is 88 chars and valid base58, but will encode to 33 bytes
+        // the leading 1s are encoded as 00 in hex and thus result in too many bytes
+        let tx_digest = "1111KKdpXH2QMB5Jm11YR48cLqUJb9Cwq2YL3tveVTPeFkZaLP8cdcH5UphVPJ7kYwCUCRLnywd3xkUhb4ZYWtf5";
+        let event_index = random_event_index();
+        let msg_id = format!("{}-{}", tx_digest, event_index);
+
+        assert!(REGEX.captures(&msg_id).is_some());
+        let res = Base58SolanaTxDigestAndEventIndex::from_str(&msg_id);
+        assert!(res.is_err());
+
+        // this is 44 chars and valid base 58, but will encode to 33 bytes
+        // (z is the largest base58 digit, and so this will overflow 2^256)
+        let tx_digest = "zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz";
+        assert_eq!(tx_digest.len(), 44);
```

Comment:
This should be adjusted to test the upper `88` char limit rather than `44`

Reply:
Also fixed at https://github.com/eigerco/axelar-amplifier/commit/c8d63449ead62eca85a14f8ac89f6f1895671955